tools/bin/ecflow/processes/main/post_processing/swe_plot.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python
+###### plotting swe percentiles
+###### usage: <python> <swe_plot.py> <configuration.cfg>
+
+###### Creates a SWE percential spatial plot 
+###### Compares one day of VIC output to 30 years of data
+###### Using the Weibull Plotting Position
+
+from datetime import datetime, timedelta
+import numpy as np
+import pandas as pd
+import xarray as xr
+import os
+import time
+import math
+import gc
+import argparse
+from netCDF4 import Dataset
+import matplotlib
+matplotlib.use('Agg') 
+import matplotlib.pyplot as plt
+from scipy.interpolate import interp1d
+from matplotlib.colors import ListedColormap
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+from cartopy.mpl.gridliner import LONGITUDE_FORMATTER, LATITUDE_FORMATTER
+import cartopy.io.shapereader as shpreader
+from tonic.io import read_config
+
+
+
+#read in configuration file
+parser = argparse.ArgumentParser(description='Reorder dimensions')
+parser.add_argument('config_file', metavar='config_file', type=argparse.FileType('r'), nargs=1, help='configuration file')
+args = parser.parse_args()
+config_dict = read_config(args.config_file[0].name)
+
+#read in from configuration file
+direc = config_dict['VIC2NC']['OutputDirNC']
+N = config_dict['ECFLOW']['Met_Delay']
+cdf_loc = config_dict['PLOT']['cdf_SWE']
+plot_loc = config_dict['PLOT']['plot_SWE']
+
+N = int(N)
+#how many days behind metdata is from realtime
+
+date_unformat = datetime.now() - timedelta(days=N)
+
+date = date_unformat.strftime('%Y-%m-%d')
+date_ncfile = date_unformat.strftime('%Y%m%d')
+month_day = date_unformat.strftime("%B_%-d")
+
+
+
+#read VIC output that has been converted to a netcdf file 
+file = "pnw_daily_run.%s-%s.nc" %(date_ncfile, date_ncfile)
+
+ds = xr.open_dataset(os.path.join(direc,file))
+
+
+#create list of all latitudes and longitudes for full rectangle
+un_lat = ds['lat'].values
+un_lon = ds['lon'].values
+
+num_lat = len(un_lat)
+num_lon = len(un_lon)
+
+latitude = np.repeat(un_lat,num_lon)
+longitude = np.tile(un_lon, num_lat)
+
+
+#Weibull Plotting Position
+q = []
+
+for i in range(1,151):
```

Comment:
Is 150 a number that could change? Should it be in a config file or taken from the length of a dataset?